pkg/controller/operators/operator_controller_test.go

```diff
@@ -43,9 +44,54 @@ var _ = Describe("Operator Controller", func() {
 		Expect(k8sClient.Create(ctx, operator)).To(Succeed())
 	})
 
-	AfterEach(func() {
-		Expect(k8sClient.Get(ctx, name, operator)).To(Succeed())
-		Expect(k8sClient.Delete(ctx, operator, deleteOpts)).To(Succeed())
+	Describe("operator deletion", func() {
+		JustBeforeEach(func() {
+			Expect(k8sClient.Delete(ctx, operator)).To(Succeed())
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, name, operator)
+				return apierrors.IsNotFound(err)
+			}, timeout, interval).Should(BeTrue())
+		})
+		Context("with components bearing its label", func() {
+			var (
+				objs      []runtime.Object
+				namespace string
+			)
+
+			BeforeEach(func() {
+				namespace = genName("ns-")
+				objs = testobj.WithLabel(expectedKey, "",
+					testobj.WithName(namespace, &corev1.Namespace{}),
+				)
+
+				for _, obj := range objs {
+					Expect(k8sClient.Create(ctx, obj.(client.Object))).To(Succeed())
+				}
+			})
+
+			AfterEach(func() {
+				for _, obj := range objs {
+					Expect(k8sClient.Get(ctx, testobj.NamespacedName(obj), obj.(client.Object))).To(Succeed())
```

Comment:
Is this `Get` call necessary? Doesn't `Delete` return an error if the resource doesn't exist?

Reply:
It shouldn't be necessary to do the `Get`, I don't think. This was a copy paste from existing test code. I'll remove and see if tests still pass.